app/cosi/cosi.go

```diff
@@ -98,139 +99,144 @@ func main() {
 
 // checkConfig contacts all servers and verifies if it receives a valid
 // signature from each.
-func checkConfig(c *cli.Context) {
-	tomlFileName := c.GlobalString("servers")
+func checkConfig(c *cli.Context) error {
+	tomlFileName := c.GlobalString(cothorityDef)
 	f, err := os.Open(tomlFileName)
-	handleErrorAndExit("Couldn't open server-file", err)
+	printErrAndExit("Couldn't open group definition file: %v", err)
 	el, err := config.ReadGroupToml(f)
-	handleErrorAndExit("Error while reading server-file", err)
+	printErrAndExit("Error while reading group definition file: %v", err)
 	if len(el.List) == 0 {
-		handleErrorAndExit("Empty entity list",
-			errors.New("Empty or invalid cosi group file:"+
-				tomlFileName))
+		printErrAndExit("Empty entity or invalid group defintion in: %s",
+			tomlFileName)
+	}
+	var wg sync.WaitGroup
+	// quick and dirty way to sum up the delat for the wait group:
+	wg.Add(len(el.List))
+	for i := range el.List {
+		for _ = range el.List[i+1:] {
+			// two calls of checkList (see below)
+			wg.Add(2)
+		}
 	}
+	fmt.Print("Will check the availability and responsiveness of the " +
+		"servers forming the group and inform you about possible " +
+		"problems.\nThis make take some time ...")
 	// First check all servers individually
 	for i := range el.List {
-		checkList(sda.NewEntityList(el.List[i : i+1]))
+		go checkList(sda.NewEntityList(el.List[i:i+1]), &wg)
 	}
 	if len(el.List) > 1 {
 		// Then check pairs of servers
 		for i, first := range el.List {
 			for _, second := range el.List[i+1:] {
 				es := []*network.Entity{first, second}
-				checkList(sda.NewEntityList(es))
+				go checkList(sda.NewEntityList(es), &wg)
 				es[0], es[1] = es[1], es[0]
-				checkList(sda.NewEntityList(es))
+				go checkList(sda.NewEntityList(es), &wg)
```

Comment:
If the master says so - merge at least the two for-loops and put the `wg.Add(2)` before line 133.